### PR TITLE
Added simple aiming system

### PR DESCRIPTION
**Pull request by Coakim**
Simple aiming system

### CODE REVIEW

File: source/dagger/gameplay/common/aiming_system.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+#include "core/graphics/sprite.h"
+#include "core/input/inputs.h"
+
+#include <cmath>
+#include <math.h>
+
+void AimingSystem::OnInitialize(Registry& registry_, Entity entity_)
+{
+    InputReceiver& receiver = registry_.get<InputReceiver>(entity_);
```
Comment (DPrivalov) [inline source/dagger/gameplay/common/aiming_system.cpp:13]: @gavrilovmiroslav , do we still need this?

File: source/dagger/gameplay/common/aiming_system.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        receiver.values[command] = 0;
+    }
+}
+
+void AimingSystem::Run()
+{
+    Engine::Registry().view<InputReceiver, Sprite, Crosshair>().each(
+        [](const InputReceiver input_, Sprite& sprite_, Crosshair& crosshair_) //The sprite component with the same entity as this crosshair component is of a sprite that is used for center of rotation (like character for example)
+        {
+            Float32 rotate = input_.values.at("rotate");        //Get the input value of the rotate command - the amount the angle changes if the button is pressed
```
Comment (DPrivalov) [inline source/dagger/gameplay/common/aiming_system.cpp:40]: add a comment above with "rotate" setup from ini file, so other teams can easily set it up.

**Review by DPrivalov — APPROVED** [review]

File: source/dagger/gameplay/common/aiming_system.cpp [hunk 3]
```diff
@@ -24,6 +24,20 @@ void AimingSystem::Run()
...
+                        "actions" : [
+                      {
+                        "trigger": "KeyA",
+                        "value" : 0.05
+                      },
+                      {
+                       "trigger": "KeyD",
+                       "value" : -0.05
+                      }
+                ]*/
```
Comment (DPrivalov) [inline source/dagger/gameplay/common/aiming_system.cpp]: */ - move this to new line

File: source/dagger/gameplay/common/aiming_system.cpp [hunk 4]
```diff
@@ -24,6 +24,20 @@ void AimingSystem::Run()
         {
             Float32 rotate = input_.values.at("rotate");        //Get the input value of the rotate command - the amount the angle changes if the button is pressed 
 
+        /*    
+        *            Example setup of for "rotate" command in input-context
```
Comment (DPrivalov) [inline source/dagger/gameplay/common/aiming_system.cpp]: remove *, and move to the left a bit, like this:

Example setup of for "rotate" command in input-context:

**Review by DPrivalov — APPROVED** [review]
there is one minor thing and good to go

File: source/dagger/gameplay/common/aiming_system.cpp (hunk 4 above)
Comment (DPrivalov) [reply source/dagger/gameplay/common/aiming_system.cpp]: Also u can move this comment before this line:

 Float32 rotate = input_.values.at("rotate");

Comment (Coakim) [reply source/dagger/gameplay/common/aiming_system.cpp]: Will do 👍

File: source/dagger/gameplay/common/aiming_system.cpp (hunk 1 above)
Comment (UbiBelETF) [reply source/dagger/gameplay/common/aiming_system.cpp:13]: No, this was changed recently, in the same event that broke Mac. We no longer need this, but the `Get` function needs to be used from `InputReceiver`. So: no setting commands to 0 manually here

File: source/dagger/gameplay/common/aiming_system.cpp [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                       "trigger": "KeyA",
+                       "value" : 0.05
+                     },
+                     {
+                      "trigger": "KeyD",
+                      "value" : -0.05
+                     }
+                    ]
+       */
+            Float32 rotate = input_.values.at("rotate");        //Get the input value of the rotate command - the amount the angle changes if the button is pressed
```
Comment (UbiBelETF) [inline source/dagger/gameplay/common/aiming_system.cpp:40]: Use `input_.Get("rotate")` instead. Maybe the `values` map should be private in the future...

**Pull request merged**